cranelift-wasm/src/environ/dummy.rs

```diff
@@ -52,6 +52,17 @@ pub struct DummyModuleInfo {
     /// Target description relevant to frontends producing Cranelift IR.
     config: TargetFrontendConfig,
 
+    /// The raw Wasm type signatures for the module being translated. This is
+    /// necessary to keep around because multi-value blocks outline their params
+    /// and returns to signatures in the type section.
+    ///
+    /// TODO FITZGEN: should this be part of `signatures` or a secondary map or
+    /// something? `signatures` is a subset of `wasm_signatures`, where we won't
+    /// have any `signatures` entry for a wasm type signature that is only used
+    /// for multi-value blocks.
```

Comment:
One thing that occurs to me is that we can move `wasm_signatures` out of the trait implementation, and into cranelift-wasm proper, because all users of cranelift-wasm will need the same thing.

We could actually then use that to construct `signatures` entries lazily, so that it indeed does only hold those signatures used by functions.

Reply:
Do you have an opinion about where the right place to put this is? The `TranslationState`?

Reply:
Yes, `TranslationState` sounds right.